### PR TITLE
feat: add generate config loading

### PR DESCRIPTION
**Pull request by katcipis**
Adds:

* Parsing of terramate.config.generate
* Hierarchical load of generate config

Integration on actual code generation (and check) will be done on a separate PR, should be considerably smaller since this is the core logic and the other functions will only get the loaded config as a parameter.

### CODE REVIEW

**Review by mariux — COMMENTED** [review]
hey @katcipis... can you give a short demo to me tomorrow about the changes? really interested how this works ... thanks a lot..

File: generate/config.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	if _, err := os.Stat(configfile); err != nil {
+		if os.IsNotExist(err) {
+			logger.Trace().Msg("config file not found.")
+			return loadStackCfg(root, filepath.Dir(configdir))
+		}
+		return StackCfg{}, fmt.Errorf("stat config file: %v", err)
+	}
+
+	logger.Trace().Msg("Read config file.")
```
Comment (i4ki) [inline generate/config.go:80]: isn't possible to use this? https://github.com/mineiros-io/terramate/blob/main/config/config.go#L64

File: hcl/hcl.go [hunk 2]
```diff
@@ -690,6 +718,65 @@ func parseRootConfig(cfg *RootConfig, block *hclsyntax.Block) error {
...
+				logger.Trace().Msg("parsing")
+
+				if attrVal.Type() != cty.String {
+					return GenerateConfig{}, fmt.Errorf("terramate.config.generate.%s is not a string but %q",
+						name,
+						attrVal.Type().FriendlyName())
+				}
+				cfg.BackendCfgFilename = attrVal.AsString()
+
+				logger.Trace().Msg("parsed with success")
```
Comment (i4ki) [inline hcl/hcl.go:756]: nice =)

**Review by i4ki — APPROVED** [review]
lgtm

**Comment by katcipis** [comment]
> hey @katcipis... can you give a short demo to me tomorrow about the changes? really interested how this works ... thanks a lot..

Not yet because this PR is the core logic but it is not integrated to the code generation itself, but integrating should be fast (but not done yet). After the follow up PR we are good for demo time (when I make changes on the cli level I always add some demo script, usually very simple but with some demo value =P).

File: generate/config.go (hunk 1 above)
Comment (katcipis) [reply generate/config.go:80]: hmm I think so, I missed the try function thing, let me give it a shot

Comment (katcipis) [reply generate/config.go]: MUCH better, thanks for pointing this out man, great catch :heart:

File: generate/config.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		return StackCfg{
+			BackendCfgFilename: BackendCfgFilename,
+			LocalsFilename:     LocalsFilename,
+		}, nil
+	}
+
+	logger.Trace().Msg("loading config.")
+
+	parsedConfig, found, err := config.TryLoadRootConfig(configdir)
+
```
Comment (i4ki) [inline generate/config.go:66]:
```suggestion
```

**Review by i4ki — APPROVED** [review]
lgtm

**Review by i4ki — APPROVED** [review]

**Pull request merged**